diff --git a/lib/_tsc.js b/lib/_tsc.js
index 612a1f7eed0fc030660b769480455aabe77b82c2..faf4d4b7b38c9785f4770c0d11e0b55d13120a31 100644
--- a/lib/_tsc.js
+++ b/lib/_tsc.js
@@ -62552,7 +62552,10 @@ function createTypeChecker(host) {
     let extraTypes;
     let tailCount = 0;
     while (true) {
-      if (tailCount === 1e3) {
+      const tailCount__LIMIT = 1e5;
+      if (tailCount === tailCount__LIMIT) {
+        const newDiag = diag(2589, 1 /* Error */, "Type_instantiation_is_excessively_deep_and_possibly_infinite_2589", `ENCOUNTERED TYPE INSTANTIATION ERROR: ${JSON.stringify({tailCount, tailCount__LIMIT})}`);
+        error(currentNode, newDiag)
         error(currentNode, Diagnostics.Type_instantiation_is_excessively_deep_and_possibly_infinite);
         return errorType;
       }
@@ -63591,8 +63594,12 @@ function createTypeChecker(host) {
     if (!couldContainTypeVariables(type)) {
       return type;
     }
-    if (instantiationDepth === 100 || instantiationCount >= 5e6) {
+    const instantiationDepth_LIMIT = 700;
+    const instantiationCount__LIMIT = 5e7;
+    if (instantiationDepth === instantiationDepth_LIMIT || instantiationCount >= instantiationCount__LIMIT) {
       (_a = tracing) == null ? void 0 : _a.instant(tracing.Phase.CheckTypes, "instantiateType_DepthLimit", { typeId: type.id, instantiationDepth, instantiationCount });
+      const newDiag = diag(2589, 1 /* Error */, "Type_instantiation_is_excessively_deep_and_possibly_infinite", `ENCOUNTERED TYPE INSTANTIATION ERROR: ${JSON.stringify({instantiationDepth, instantiationDepth_LIMIT, instantiationCount, instantiationCount__LIMIT})}`);
+      error(currentNode, newDiag)
       error(currentNode, Diagnostics.Type_instantiation_is_excessively_deep_and_possibly_infinite);
       return errorType;
     }
diff --git a/lib/typescript.js b/lib/typescript.js
index 0554fc3fc707ce3edbc3c4f8f4d77f8aa3def7ba..7bf9ef736b65d01807548358c1c3eed10db64492 100644
--- a/lib/typescript.js
+++ b/lib/typescript.js
@@ -67163,7 +67163,10 @@ function createTypeChecker(host) {
     let extraTypes;
     let tailCount = 0;
     while (true) {
-      if (tailCount === 1e3) {
+      const tailCount__LIMIT = 1e5;
+      if (tailCount === tailCount__LIMIT) {
+        const newDiag = diag(2589, 1 /* Error */, "Type_instantiation_is_excessively_deep_and_possibly_infinite_2589", `ENCOUNTERED TYPE INSTANTIATION ERROR: ${JSON.stringify({tailCount, tailCount__LIMIT})}`);
+        error2(currentNode, newDiag)
         error2(currentNode, Diagnostics.Type_instantiation_is_excessively_deep_and_possibly_infinite);
         return errorType;
       }
@@ -68202,8 +68205,12 @@ function createTypeChecker(host) {
     if (!couldContainTypeVariables(type)) {
       return type;
     }
-    if (instantiationDepth === 100 || instantiationCount >= 5e6) {
+    const instantiationDepth_LIMIT = 700;
+    const instantiationCount__LIMIT = 5e7;
+    if (instantiationDepth === instantiationDepth_LIMIT || instantiationCount >= instantiationCount__LIMIT) {
       (_a = tracing) == null ? void 0 : _a.instant(tracing.Phase.CheckTypes, "instantiateType_DepthLimit", { typeId: type.id, instantiationDepth, instantiationCount });
+      const newDiag = diag(2589, 1 /* Error */, "Type_instantiation_is_excessively_deep_and_possibly_infinite_2589", `ENCOUNTERED TYPE INSTANTIATION ERROR: ${JSON.stringify({instantiationDepth, instantiationDepth_LIMIT, instantiationCount, instantiationCount__LIMIT})}`);
+      error2(currentNode, newDiag)
       error2(currentNode, Diagnostics.Type_instantiation_is_excessively_deep_and_possibly_infinite);
       return errorType;
     }
